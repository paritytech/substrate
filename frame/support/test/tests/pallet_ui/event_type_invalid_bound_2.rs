#[frame_support::pallet]
mod pallet {
	use frame_support::pallet_prelude::{Hooks, IsType};
	use frame_system::pallet_prelude::BlockNumberFor;

	#[pallet::config]
	pub trait Config: frame_system::Config {
		type Bar;
		type RuntimeEvent: IsType<<Self as frame_system::Config>::RuntimeEvent>;
	}

	#[pallet::pallet]
	pub struct Pallet<T>(core::marker::PhantomData<T>);

	#[pallet::hooks]
	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}

	#[pallet::call]
	impl<T: Config> Pallet<T> {}

	#[pallet::event]
	pub enum Event<T: Config> {
		B { b: T::Bar },
	}
}

fn main() {
}
