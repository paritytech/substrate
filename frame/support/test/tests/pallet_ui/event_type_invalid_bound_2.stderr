error: Invalid `type RuntimeEvent`, associated type `RuntimeEvent` is reserved and must bound: `From<Event>` or `From<Event<Self>>` or `From<Event<Self, I>>`
 --> $DIR/event_type_invalid_bound_2.rs:9:3
  |
9 |         type RuntimeEvent: IsType<<Self as frame_system::Config>::RuntimeEvent>;
  |         ^^^^
